webrender/src/device/gfx/buffer.rs

```diff
@@ -268,93 +272,120 @@ impl<B: hal::Backend> InstancePoolBuffer<B> {
         self.last_update_size = 0;
     }
 
-    fn deinit(self, device: &B::Device, heaps: &mut Heaps<B>) {
+    pub(super) fn deinit(self, device: &B::Device, heaps: &mut Heaps<B>) {
         self.buffer.deinit(device, heaps);
     }
+
+    fn free(&self) -> usize {
+        self.buffer.buffer_size - self.offset
+    }
+
+    fn can_store_data(&self, stride: usize) -> bool {
+        if self.offset_aligned_to(stride) {
+            self.free() >= stride
+        } else {
+            let offset = self.offset + (stride - (self.offset % stride));
```

Comment:
there is some redundant work going on here: we just called `offset_aligned_to`, which does the same `%` operation

Reply:
Reworked the logic, now we compute the next suitable offset once: https://github.com/szeged/webrender/pull/300/commits/d5c90a5aaa484f69dfdbbddacc74449a7ffa6b60#diff-5dbe2f57bb9273f9a363bcb71ec6834eR284